C#/Botkit/Microsoft.BotKit/BotWorker.cs

```diff
@@ -123,25 +146,131 @@ public async Task<DialogTurnResult> CancelAllDialogs()
         /// </summary>
         /// <param name="id">ID of dialog</param>
         /// <param name="options">Object containing options to be passed into the dialog</param>
-        public async Task ReplaceDialog(string id, object options)
+        public async Task ReplaceDialogAsync(string id, Dictionary<string, object> options)
         {
+            if (config.DialogContext != null)
```

Comment:
Can we apply here a similar approach as we previously did?

Reply:
Done